service/matching/taskQueueManager.go

```diff
@@ -457,51 +459,57 @@ func (c *taskQueueManagerImpl) DispatchQueryTask(
 // has no cached data, it will explicitly attempt a fetch from the root partition.
 func (c *taskQueueManagerImpl) GetVersioningData(ctx context.Context) (*persistencespb.VersioningData, error) {
 	vd, err := c.db.GetVersioningData(ctx)
-	if errors.Is(err, &versioningDataNotPresentOnSubPartition{}) {
-		// If this is a sub-partition with no versioning data, this call is indicating we might expect to find
+	if errors.Is(err, errVersioningDataNotPresentOnPartition) {
+		// If this is a non-root-partition with no versioning data, this call is indicating we might expect to find
 		// some. Since we may not have started the poller, we want to explicitly attempt to fetch now, because
-		// it's possible some was added to the root partition, but it failed to invalidate this sub partition.
+		// it's possible some was added to the root partition, but it failed to invalidate this partition.
 		return c.fetchMetadataFromRootPartition(ctx)
 	}
 	return vd, err
 }
 
 func (c *taskQueueManagerImpl) MutateVersioningData(ctx context.Context, mutator func(*persistencespb.VersioningData) error) error {
-	err := c.db.MutateVersioningData(ctx, mutator)
-	// If this is the root partition, notify sub partitions that they should fetch changed data from us.
+	newDat, err := c.db.MutateVersioningData(ctx, mutator)
+	c.signalIfFatal(err)
+	if err != nil {
+		return err
+	}
+	// If this is the root partition, notify partitions that they should fetch changed data from us.
 	if c.taskQueueID.IsRoot() {
```

Comment:
maybe more early return?
```suggestion
	if !c.taskQueueID.IsRoot() {
	  return nil
	}
	// rest of code here
	return nil
```